docs/standard/collections/when-to-use-generic-collections.md

```diff
@@ -1,51 +1,67 @@
-  
+Using generic collections gives you the automatic benefit of type safety without having to derive from a base collection type and implement type-specific members. Generic collection types also generally perform better than the corresponding nongeneric collection types (and better than types that are derived from nongeneric base collection types) when the collection elements are value types, because with generics, there's no need to box the elements.
+
+For programs that target .NET Standard 1.0 or later, use the generic collection classes in the <xref:System.Collections.Concurrent> namespace when multiple threads might be adding or removing items from the collection concurrently. Additionally, when immutability is desired, consider the generic collection classes in the <xref:System.Collections.Immutable> namespace.
+
+The following generic types correspond to existing collection types:
+
+- <xref:System.Collections.Generic.List%601> is the generic class that corresponds to <xref:System.Collections.ArrayList>.
+
+- <xref:System.Collections.Generic.Dictionary%602> and <xref:System.Collections.Concurrent.ConcurrentDictionary%602> are the generic classes that correspond to <xref:System.Collections.Hashtable>.
+
+- <xref:System.Collections.ObjectModel.Collection%601> is the generic class that corresponds to <xref:System.Collections.CollectionBase>. <xref:System.Collections.ObjectModel.Collection%601> can be used as a base class, but unlike <xref:System.Collections.CollectionBase>, it is not abstract, which makes it much easier to use.
+
+- <xref:System.Collections.ObjectModel.ReadOnlyCollection%601> is the generic class that corresponds to <xref:System.Collections.ReadOnlyCollectionBase>. <xref:System.Collections.ObjectModel.ReadOnlyCollection%601> is not abstract and has a constructor that makes it easy to expose an existing <xref:System.Collections.Generic.List%601> as a read-only collection.
+
+- The <xref:System.Collections.Generic.Queue%601>, <xref:System.Collections.Concurrent.ConcurrentQueue%601>, <xref:System.Collections.Immutable.ImmutableQueue%601>, <xref:System.Collections.Generic.Array%601>, <xref:System.Collections.Concurrent.ConcurrentArray%601>, <xref:System.Collections.Immutable.ImmutableArray%601>, <xref:System.Collections.Generic.SortedList%602>, and <xref:System.Collections.Immutable.ImmutableSortedSet%601> generic classes correspond to the respective nongeneric classes with the same names.
```

Comment:
I am not finding Generic.Array<T> or Concurrent.ConcurrentArray<T>. Are you sure those types exist?

Reply:
I have removed them